src/System.Memory/src/System/Buffers/ReadOnlyBuffer_helpers.cs

```diff
@@ -0,0 +1,305 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    public readonly partial struct ReadOnlyBuffer<T>
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I don't think this should be marked with AggressiveInlining. The method body is too large.